pac4j-http/src/main/java/org/pac4j/http/client/BasicAuthClient.java

```diff
@@ -0,0 +1,25 @@
+/*
+  Copyright 2012 - 2015 pac4j organization
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+       http://www.apache.org/licenses/LICENSE-2.0
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+ */
+package org.pac4j.http.client;
+
+import org.pac4j.http.client.indirect.IndirectBasicAuthClient;
+
+/**
+ * Will be removed. Use {@link IndirectBasicAuthClient} instead.
```

Comment:
This was an old class I first moved to another package and changed the behaviour radically. As I realize it is a bad idea, I recreate it empty, inheriting from the class which replaces it, to help users find the new non-deprecated class they should use.
We now have a `DirectBasicAuthClient` for stateless use cases and a `IndirectBasicAuthClient` for stateful scenarios.

Does it make sense?